runtime/src/accounts_index.rs

```diff
@@ -1194,12 +1199,21 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
                     max_clean_root,
                     account_indexes,
                 );
+                is_slot_list_empty = slot_list.is_empty();
             });
         }
+        if is_slot_list_empty {
+            let mut w_maps = self.account_maps.write().unwrap();
+            if let Some(x) = w_maps.get(pubkey) {
+                if x.slot_list.read().unwrap().is_empty() {
+                    w_maps.remove(pubkey);
+                }
+            }
+        }
     }
 
-    pub fn can_purge(max_root: Slot, slot: Slot) -> bool {
-        slot < max_root
+    fn can_purge_older_entries(max_clean_root: Slot, max_cleanable_root: Slot, slot: Slot) -> bool {
+        slot < max_clean_root && slot != max_cleanable_root
```

Comment:
this line isn't so self-explanatory.. I mean there is little aid for readers to understand the difference of those max clean root variables, only with or without `able`.. how about introducing `newest_` to avoid overloading `max_` to mean the slot is something most recent in the AccountsIndex list. this will yet again change this fn name... https://github.com/solana-labs/solana/pull/16911/files#r623961443 ;)

Then, this line would read like this:
```rust
slot < max_clean_root && slot != newest_root_in_slot_list`
```

Reply:
I like this idea. I personally found it confusing to understand the different "max" types, so I'm onboard!